source/CoordinateConversion/CoordinateConversionLibrary/Models/CoordinateMGRS.cs

```diff
@@ -57,7 +57,7 @@ public static bool TryParse(string input, out CoordinateMGRS mgrs)
             
             input = input.Trim().ToUpper();
```

Comment:
@kgonzago it seems to me that we can either convert the string to uppercase or fix the regex. Is there a reason to do both? I converted to uppercase because it was easier, but we can probably remove that since you did the work to fix the regex.